### PR TITLE
make hlut widget optional

### PR DESCRIPTION
**Pull request by clewis7**

### CODE REVIEW

**Comment by clewis7** [comment]
would be nice to add an hlut button to the toolbar so it can be generated after imagewidget instantiation if a user wanted one for a specific subplot!

File: fastplotlib/widgets/image.py [hunk 1]
```diff
@@ -288,6 +289,9 @@ def __init__(
         names: Optional[str]
             gives names to the subplots
 
+        histogram_widget: bool, default False
+            determines whether each subplot is rendered with a Histogram LUT widget
```
Comment (kushalkolar) [inline fastplotlib/widgets/image.py:293]: More concise: make histogram LUT widget for each image?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Comment by kushalkolar** [comment]
> would be nice to add an hlut button to the toolbar so it can be generated after imagewidget instantiation if a user wanted one for a specific subplot!

I think this is feature creep, we they can create and add it programmatically if they need that level of customization?

**Comment by kushalkolar** [comment]
Also add kwarg to specify subsample dims for HistogramLut widget?

File: fastplotlib/widgets/image.py [hunk 2]
```diff
@@ -225,6 +225,7 @@ def __init__(
         grid_shape: Tuple[int, int] = None,
         names: List[str] = None,
         grid_plot_kwargs: dict = None,
+        histogram_widget: bool = False,
```
Comment (kushalkolar) [inline fastplotlib/widgets/image.py:228]: I think default should be `True` :D

Comment (clewis7) [inline fastplotlib/widgets/image.py:228]:
```suggestion
        histogram_widget: bool = True,
```

File: fastplotlib/widgets/image.py (hunk 1 above)
Comment (clewis7) [inline fastplotlib/widgets/image.py:293]:
```suggestion
            make histogram LUT widget for each subplot
```

**Comment by kushalkolar** [comment]
> > Also add kwarg to specify subsample dims for HistogramLut widget?
> 
> @kushalkolar anything more than number of bins? or you want a dict that allows user to pass nbins and flank?

Just a tuple of its to specify the max size?

**Comment by kushalkolar** [comment]
going to merge this, see #359 for the subsample

**Pull request merged**